source/ftrack_connect_pipeline_qt/ui/utility/widget/asset_version_list_selector.py

```diff
@@ -146,10 +154,24 @@ def _query_assets_from_context_async(self, context_id, asset_type_name):
         assets = self.session.query(
             'select name, versions.task.id, type.id, id, latest_version,'
             'latest_version.version, latest_version.date '
-            'from Asset where versions.task.id is {} and type.id is {}'.format(
+            'from Asset where (versions.task.id is {0} or versions.asset.parent.id is {0}) and type.id is {1}'.format(
                 context_id, asset_type_entity['id']
             )
         ).all()
+        # Follow context incoming links
+        for tcl in self.session.query(
+            'TypedContextLink where to_id is "{}"'.format(context_id)
+        ):
+            for asset in self.session.query(
+                'select name, versions.task.id, type.id, id, latest_version,'
+                'latest_version.version, latest_version.date '
+                'from Asset where versions.asset.parent.id is {} and type.id is {}'.format(
+                    tcl['from_id'], asset_type_entity['id']
+                )
+            ).all():
+                if not asset['id'] in [_asset['id'] for _asset in assets]:
+                    assets.append(asset)
+
```

Comment:
All this should happen on a opener plugin or in pipeline, but not in QT. So I should be able to open the correct asset in standalone without the need of having QT.

Reply:
Will add a TODO so we can revisit this later and move this logic out of QT.